### PR TITLE
drop symfony 4.4 support

### LINKED ISSUE #171: symfony/symfony-templating set as dev requirement
symfony/symfony-templating is set as dev requirement, but the TimeHelper class is depending on it

### PR DESCRIPTION
**Pull request by jrushlow**
- Bumps minimum Symfony version to `5.4`
- refactors bundle to use Symfony 5.x bundle standards.
- removes legacy code
- fixes #171

### CODE REVIEW

File: src/KnpTimeBundle.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Knp\Bundle\TimeBundle;
+
+use Knp\Bundle\TimeBundle\DependencyInjection\KnpTimeBundleExtension;
+use Symfony\Component\DependencyInjection\Extension\ExtensionInterface;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+final class KnpTimeBundle extends Bundle
+{
+    public function getContainerExtension(): ?ExtensionInterface
```
Comment (weaverryan) [inline src/KnpTimeBundle.php:11]: Why is this needed? If it were called `KnpTimeExtension`, wouldn't this method be not needed?

File: composer.json [hunk 2]
```diff
@@ -51,5 +56,10 @@
         "branch-alias": {
             "dev-master": "1.1.x-dev"
         }
+    },
+    "config": {
+        "allow-plugins": {
+            "symfony/flex": true
+        }
```
Comment (weaverryan) [inline composer.json:63]: Why was this needed? It looks like we already allow in CI

Comment (jrushlow) [reply composer.json]: good catch

File: src/KnpTimeBundle.php (hunk 1 above)
Comment (jrushlow) [reply src/KnpTimeBundle.php]: Learn something new everyday

File: composer.json [hunk 3]
```diff
@@ -19,16 +19,16 @@
 
     "require": {
         "php":                          ">=7.2.5",
-        "symfony/dependency-injection": "^4.3|^5.0|^6.0",
-        "symfony/translation":          "^4.3|^5.0|^6.0",
-        "symfony/config":               "^4.3|^5.0|^6.0"
+        "symfony/dependency-injection": "^5.4|^6.0",
+        "symfony/templating": "^5.4|^6.0",
+        "symfony/translation":          "^5.4|^6.0",
+        "symfony/config":               "^5.4|^6.0"
```
Comment (weaverryan) [inline composer.json:25]: we have alignment in some places but not others - let's ditch the alignment everywhere

**Review by weaverryan — APPROVED** [review]
Looks good!
